device/blackhole/blackhole_implementation.cpp

```diff
@@ -2,11 +2,11 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#include "blackhole_implementation.h"
+#include "umd/device/blackhole_implementation.h"
 
-#include "device/cluster.h"
-#include "src/firmware/riscv/blackhole/eth_interface.h"
-#include "src/firmware/riscv/blackhole/host_mem_address_map.h"
+#include "blackhole/eth_interface.h"
```

Comment:
This is src/firmware/risv/blackhole/eth_interface.h ?

But the src/firmware/riscv is not exposed publicly, right?

Reply:
Define 'public' :)

`src/firmware/riscv` is public from [firmware](https://github.com/tenstorrent/tt-umd/blob/afuller/clarify-api/src/CMakeLists.txt#L6)
Firmware is private to [device](https://github.com/tenstorrent/tt-umd/blob/afuller/clarify-api/device/CMakeLists.txt#L53).

So like a Foo that has a private member variable Bar, Bar's public interface is not exposed through Foo.  But anyone who uses Bar directly can see it.
umd::device gets to see `src/firmware/riscv`, but anyone who uses umd::device will NOT see `src/firmware/riscv`.